CDCgov/prime-reportstream
# Fetch file content for download

## Problem
When the download report button is clicked, the file created is empty. This is because when the report data is fetched, server intentionally leaves out the content for performance reasons. We have to run another fetch to get the csv content. 

## Solution
When the button is clicked, fetch `/api/history/report/${reportId}` and download the content

## Test Steps
- From daily-data page, download a report and verify it is not empty
- From the report page, download a report and verify it is not empty

## Checklist

### Testing

### Process

## Fixes
- https://github.com/CDCgov/prime-reportstream/issues/2610

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints? No
- Does this PR include changes in authentication and/or authorization of existing endpoints? No
- Does this change introduce new dependencies that need vetting? No
- Does this change require changes to our infrastructure? No
- Does logging contain sensitive data? No
- Does this PR include or remove any sensitive information itself? No

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
